blockchain/state_manager/src/lib.rs

```diff
@@ -122,6 +138,83 @@ where
         Ok((state_root, rect_root))
     }
 
+    pub async fn tipset_state(&self, tipset: &Tipset) -> Result<(Cid, Cid), Box<dyn StdError>> {
+        span!("tipset_state", {
+            trace!("tipset {:?}", tipset.cids());
+            // if exists in cache return
+            if let Some(cid_pair) = self.cache.read().await.get(&tipset.key()) {
+                return Ok(cid_pair.clone());
+            }
+
+            if tipset.len() == 0 {
```

Comment:
This should be checking if the tipset height is ==0, no?

Reply:
No height method on Tipset struct. Do you mean weight?

Reply:
Sorry I mean epoch, I use those synonymously